src/Token/AccessToken.php

```diff
@@ -100,6 +100,21 @@ public function getUid()
     }
 
     /**
+     * Checks if the token has expired.
+     *
+     * @return boolean true if the token has expired, false otherwise.
+     * @throws InvalidArgumentException if the expires is not set on the token.
+     */
+    public function hasExpired()
+    {
+        if (!isset($this->expires)) {
+            throw new InvalidArgumentException('"expires" is not set on the token');
```

Comment:
I gave that some thought, too, but I wasn't sure what would best fit. I think you're right: `RuntimeException` would be best.